### PR TITLE
SPCTR-2444-Improvement: Code: optimization of Spectra-settings functions in common-setting.php

### PR DESCRIPTION
**Pull request by rajmohanreddy774**
### Description
code-optimization of Common_Settings class functions.

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: admin-core/ajax/common-settings.php [hunk 1]
```diff
@@ -87,6 +87,34 @@ public function register_ajax_events() {
 
 		$this->init_ajax_events( $ajax_events );
 	}
+
+	/**
+	 * checks value in post
+	 *
+	 * @return void
+	 */
```
Comment (adi3890) [inline admin-core/ajax/common-settings.php:95]: function comment improper should have all params detail

File: admin-core/ajax/common-settings.php [hunk 2]
```diff
@@ -87,6 +87,34 @@ public function register_ajax_events() {
...
+	private function check_value_in_post( $arr, $key, $error_msg = false ) {
+		$response_data = $error_msg ? $error_msg : array( 'messsage' => __( 'No post data found!', 'ultimate-addons-for-gutenberg' ) );
+		if ( ! isset( $arr [ $key ] ) ) {
```
Comment (adi3890) [inline admin-core/ajax/common-settings.php:98]: we need only empty check

File: admin-core/ajax/common-settings.php [hunk 3]
```diff
@@ -87,6 +87,34 @@ public function register_ajax_events() {
...
+		} elseif ( empty( $arr [ $key ] ) ) {
+			wp_send_json_error( $response_data );
+		}
+	}
+
+	/**
+	 * save and success response
+	 *
+	 * @return void
+	 */
```
Comment (adi3890) [inline admin-core/ajax/common-settings.php:109]: incomplete doc comment

File: admin-core/ajax/common-settings.php [hunk 4]
```diff
@@ -87,6 +87,34 @@ public function register_ajax_events() {
...
+	private function save_and_send_success_response( $admin_settings_option, $value ) {
```
Comment (adi3890) [inline admin-core/ajax/common-settings.php:110]: we can have simple variable name $option
as this is in scope of function.

sanitize data

File: admin-core/ajax/common-settings.php [hunk 5]
```diff
@@ -154,20 +182,8 @@ public function recaptcha_secret_key_v3() {
...
-		if ( isset( $_POST['value'] ) ) {
-			\UAGB_Admin_Helper::update_admin_settings_option( 'uag_recaptcha_secret_key_v3', sanitize_text_field( $_POST['value'] ) );
-		}
-
-		$response_data = array(
-			'messsage' => __( 'Successfully saved data!', 'ultimate-addons-for-gutenberg' ),
-		);
-		wp_send_json_success( $response_data );
-
+		$this->check_value_in_post( $_POST, 'value' );
```
Comment (adi3890) [inline admin-core/ajax/common-settings.php:185]: create one function

File: admin-core/ajax/common-settings.php (hunk 1 above)
Comment (rajmohanreddy774) [reply admin-core/ajax/common-settings.php:98]: Added params

File: admin-core/ajax/common-settings.php (hunk 3 above)
Comment (rajmohanreddy774) [reply admin-core/ajax/common-settings.php:112]: Added complete comment

File: admin-core/ajax/common-settings.php (hunk 2 above)
Comment (rajmohanreddy774) [reply admin-core/ajax/common-settings.php]: done changes

File: admin-core/ajax/common-settings.php (hunk 4 above)
Comment (rajmohanreddy774) [reply admin-core/ajax/common-settings.php]: done changes

File: admin-core/ajax/common-settings.php (hunk 5 above)
Comment (rajmohanreddy774) [reply admin-core/ajax/common-settings.php:188]: can you pls elaborate?

**Comment by RajkiranBagal** [comment]
@rajmohanreddy774  
- PR title not valid
 - Code analysis in PR

**Comment by rajmohanreddy774** [comment]
> @rajmohanreddy774
> 
> * PR title not valid
> * Code analysis in PR

@RajkiranBagal 
code analysis in PR is passed.
PR title changed.

**Comment by RajkiranBagal** [comment]
@rajmohanreddy774 PR title still not proper. As per [this](https://docs.google.com/document/d/1oiF7iFLSx5GQprXhnLjwdVcHK7dFwaCW1-qqOxcs1CA/edit)

**Review by rahulvarma722 — APPROVED** [review]

File: admin-core/ajax/common-settings.php [hunk 6]
```diff
@@ -87,6 +87,65 @@ public function register_ajax_events() {
...
+	 * Saves the success message after successfully updating admin settings option.
+	 *
+	 * @param string $option The name of the option to update.
+	 * @param string $value The value to be updated.
+	 * @since x.x.x
+	 * @return void
+	 */
+	public function save_admin_settings( $option, $value ) {
+		// nonce verification done in function check_permission_nonce.
+		if ( isset( $_POST['value'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply admin-core/ajax/common-settings.php]: Removed and Tested works fine.

File: admin-core/ajax/common-settings.php [hunk 7]
```diff
@@ -87,6 +87,65 @@ public function register_ajax_events() {
...
+	public function check_permission_nonce( $option, $scope = 'manage_options' ) {
+
+		if ( ! current_user_can( $scope ) ) {
+			wp_send_json_error( array( 'messsage' => $this->get_error_msg( 'permission' ) ) );
+		}
+
+		/**
+		 * Nonce verification
+		 */
+		if ( ! check_ajax_referer( $option, 'security', false ) ) {
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply admin-core/ajax/common-settings.php]: Added third parameter now.

File: admin-core/ajax/common-settings.php [hunk 8]
```diff
@@ -140,34 +199,9 @@ public static function pro_activate() {
...
-			'messsage' => __( 'Successfully saved data!', 'ultimate-addons-for-gutenberg' ),
-		);
-		wp_send_json_success( $response_data );
-
+		$this->check_permission_nonce( 'uag_recaptcha_secret_key_v3' );
+		$value = $this->check_post_value();
+		$this->save_admin_settings( 'uag_recaptcha_secret_key_v3', sanitize_text_field( $value ) );
 	}
 	/**
 	 * Save settings.
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply admin-core/ajax/common-settings.php]: updated descriptions of functions.

**Comment by rajmohanreddy774** [comment]
> @rajmohanreddy774 PR title still not proper. As per [this](https://docs.google.com/document/d/1oiF7iFLSx5GQprXhnLjwdVcHK7dFwaCW1-qqOxcs1CA/edit)

--Added proper PR title as per docs

File: admin-core/ajax/common-settings.php [hunk 9]
```diff
@@ -90,6 +90,65 @@ public function register_ajax_events() {
...
+	/**
+	 * Saves the success message after successfully updating admin settings option.
+	 *
+	 * @param string $option The name of the option to update.
+	 * @param string $value The value to be updated.
+	 * @since x.x.x
+	 * @return void
+	 */
+	private function save_admin_settings( $option, $value ) {
+		// nonce verification done in function check_permission_nonce.
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply admin-core/ajax/common-settings.php]: Removed

File: admin-core/ajax/common-settings.php [hunk 10]
```diff
@@ -1278,7 +602,7 @@ public function regenerate_assets() {
 	}
 
 	/**
-	 * Save settings.
+	 * Save setting - Sanitizes form inputs.
 	 *
 	 * @param array $input_settings settimg data.
```
Comment (bsf-zanev) [inline admin-core/ajax/common-settings.php:607]: Could we add `@return` under this param?

**Review by bsf-zanev — CHANGES_REQUESTED** [review]

**Review by supriyajjadhav — APPROVED** [review]

File: admin-core/ajax/common-settings.php (hunk 10 above)
Comment (bsf-zanev) [reply admin-core/ajax/common-settings.php:613]:
```suggestion
	 * @param array $input_settings settimg data.
	 * @return array    The sanitized data form inputs.
```

Comment (rajmohanreddy774) [reply admin-core/ajax/common-settings.php:613]: done changes

**Review by bsf-zanev — APPROVED** [review]

**Review by aapatre — APPROVED** [review]

File: admin-core/ajax/common-settings.php [hunk 11]
```diff
@@ -326,467 +265,149 @@ public function fetch_pages() {
...
-		}
+	/**
+	 * Save setting - Saves selected font globally.
+	 *
+	 * @return void
+	 */
+	public function select_font_globally() {
+		$this->check_permission_nonce( 'uag_select_font_globally' );
+		$value = $this->check_post_value();
+		$value = isset( $value ) ? json_decode( stripslashes( $value ), true ) : array(); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
```
Comment (adi3890) [inline admin-core/ajax/common-settings.php:352]: do we need isset here.
please check for similar instance in this PR.

Proper comment not available for phpcs:ignore
please check for same everywhere in this file.

File: admin-core/ajax/common-settings.php [hunk 12]
```diff
@@ -1407,20 +676,10 @@ public function insta_linked_accounts() {
...
-		}
-		/**
-		 * Nonce verification
-		 */
-		if ( ! check_ajax_referer( 'uag_insta_all_users_media', 'security', false ) ) {
-			wp_send_json_error( array( 'messsage' => $this->get_error_msg( 'nonce' ) ) );
-		}
-		$value = isset( $_POST['value'] ) ? json_decode( stripslashes( $_POST['value'] ), true ) : array(); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$this->check_permission_nonce( 'uag_insta_all_users_media' );
+		$value = isset( $_POST['value'] ) ? json_decode( stripslashes( $_POST['value'] ), true ) : array(); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized,WordPress.Security.NonceVerification.Missing
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply admin-core/ajax/common-settings.php]: yes done changes

File: admin-core/ajax/common-settings.php [hunk 13]
```diff
@@ -326,467 +265,149 @@ public function fetch_pages() {
...
-			wp_send_json_error( $response_data );
-		}
-
-		if ( empty( $_POST ) ) {
-			$response_data = array( 'messsage' => __( 'No post data found!', 'ultimate-addons-for-gutenberg' ) );
-			wp_send_json_error( $response_data );
-		}
-
+		$this->check_permission_nonce( 'uag_social' );
+		$this->check_post_value();
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply admin-core/ajax/common-settings.php]: removed

File: admin-core/ajax/common-settings.php (hunk 11 above)
Comment (rajmohanreddy774) [reply admin-core/ajax/common-settings.php]: yes removed isset in similar instances,
removed phpcs:ignore that is of no use.

**Review by adi3890 — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**